services/email/app/main.py

```diff
@@ -0,0 +1,163 @@
+
+
+class InternalAuth(object):  # pylint: disable=too-few-public-methods
+    def __call__(self, req):
+        req.headers['Authorization'] = INTERNAL_COMMUNICATION_KEY
+        return req
+
+
+SESSION = requests.Session()
+SESSION.auth = InternalAuth()
+
+
+def get_task_store_uri(path: str) -> str:
+    # in debug mode, the service is likely run out of a cluster, switch to https schema
+    if app.debug:
+        return f'https://{STORE_HOST}/api/{path}'
+    return f'http://{STORE_HOST}/api/{path}'
+
+
+@app.route('/health')
+def healthy():
+    status = 'healthy'
+    remark = ''
+
+    if not SMTP_SERVER or not SMTP_USER or not SMTP_PASS:
+        logger.error('missing environment variable to setup email service.')
+        status = 'unhealthy'
+        remark = 'missing SMTP settings'
+
+    if not INTERNAL_COMMUNICATION_KEY:
+        logger.error('missing internal communication key')
+        status = 'unhealthy'
+        remark = 'missing internal communication key'
+
+    return jsonify({'status': status, 'time': datetime.utcnow(), 'remark': remark})
+
+
+@app.route('/report', methods=['POST'])
+def send_report():
+    logger.info('requested to send email')
+    run_id = request.json['run_id']
+    receivers = request.json['receivers']
+    logger.info(f'run: {run_id} | receivers: {receivers}')
+
+    run = SESSION.get(get_task_store_uri(f'run/{run_id}')).json()
+    tasks = sorted(SESSION.get(get_task_store_uri(f'run/{run_id}/tasks')).json(), key=lambda t: t['status'])
+
+    logger.info(f'successfully read run {run_id}.')
+
+    statuses = defaultdict(lambda: 0)
+    results = defaultdict(lambda: 0)
+
+    failure = []
+
+    logging.info('begin composing report')
+    for task in tasks:
+        status = task['status']
+        result = task['result']
+
+        statuses[status] = statuses[status] + 1
+        results[result] = results[result] + 1
+
+        if result != 'Passed':
+            if run['details'].get('a01.reserved.product', None) == 'azurecli':
+                test_name = task['settings']['classifier']['identifier']
+                if test_name.startswith('azure.cli.command_modules'):
+                    module_name = test_name.split('.')[3]
+                else:
+                    module_name = 'CORE'
+            else:
+                module_name = None
+
+            failure.append(
+                (task['id'],
+                 task['name'].rsplit('.')[-1],
+                 task['status'],
+                 task['result'],
+                 (task.get('result_details') or dict()).get('duration'),
+                 module_name))
+
+    status_summary = ' | '.join([f'{status_name}: {count}' for status_name, count in statuses.items()])
+    result_summary = ' | '.join([f'{result or "Not run"}: {count}' for result, count in results.items()])
+
+    creation = datetime.strptime(run['creation'], '%Y-%m-%dT%H:%M:%SZ') - timedelta(hours=8)
+
+    summaries = [('Id', run['id']),
```

Comment:
>summaries [](start = 4, length = 9)

Would it be possible to also include what `--mode` is in the client? On go run we use it to know if it is against prod or canary

Reply:
Can do.

Reply:
https://github.com/Azure/adx-automation-services/issues/2